src/seccomp/src/syscall_table.rs

```diff
@@ -0,0 +1,673 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashMap;
+
+/// Creates and owns a mapping from the arch-specific syscall name to the right number.
+#[derive(Debug)]
+pub(crate) struct SyscallTable {
+    map: HashMap<String, i64>,
+    arch: String,
+}
+
+impl SyscallTable {
+    pub fn new(arch: String) -> Self {
+        let mut instance = Self {
+            arch,
+            map: HashMap::with_capacity(332), // nr of syscalls for x86_64 (upper bound)
+        };
+
+        instance.populate_map();
+
+        instance
+    }
+
+    /// Returns the arch-specific syscall number based on the given name.
+    /// Adheres to the "SYS_*" notation.
+    pub fn get_syscall_nr(&self, sys_name: &str) -> Option<i64> {
+        self.map.get(sys_name).copied()
+    }
+
+    // We need to create an architecture-specific map since the set of
+    // possible syscalls is architecture-dependent. We can't just map to their
+    // corresponding libc constant.
+    fn populate_map(&mut self) {
+        if self.arch == "aarch64" {
+            self.map.insert("SYS_io_setup".to_string(), 0);
```

Comment:
After giving this some more thought, I don't think this is a safe thing to do.
1. Reviewers (me :fearful:) should verify that each of these hardcoded numbers does indeed correspond to the denominated syscall. That's very error prone. Further additions to these lists are prone to the same reviewer errors: a reviewer might miss an erroneous syscall and inadvertently create a security flaw.
1. Even before the review, new additions to this list could add mistakes that will cause faulty code to run as part of the CI build. This should be avoided.

I suggest generating this list either using `build.rs` (preferred, as it protects against mistakes in future PRs) or with an external tool, then copy-pasting it here, from the header files present on the system (all of these are defined somewhere in `/usr/include`, and it's a matter of parsing C headers).

Reply:
putting these syscalltable modifications on hold for now. I need to think a little more about it

Reply:
Generating these seems to bring in extra complexity and security concerns or failure points. AFAIK syscall numbers are part of the kernel ABI and don't change so why not just hardcode them in a source file?

Initial commit of the hardcoded list needs to be carefully reviewed, but then changes there (which should be rare) are few-liners easy to review.

If we're really paranoid we can add extra validation like unittests that check our hardcoded values against kernel headers or smth.

Reply:
We synced offline and decided to auto-generate the syscall tables, on-demand. I will work on writing a script for doing this, from the linux kernel source code.

Reply:
done, PTAL